src/Framework/ResultCacheExtension.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework;
+
+use PHPUnit\Runner\AfterIncompleteTestHook;
+use PHPUnit\Runner\AfterLastTestHook;
+use PHPUnit\Runner\AfterRiskyTestHook;
+use PHPUnit\Runner\AfterSkippedTestHook;
+use PHPUnit\Runner\AfterSuccessfulTestHook;
+use PHPUnit\Runner\AfterTestErrorHook;
+use PHPUnit\Runner\AfterTestFailureHook;
+use PHPUnit\Runner\AfterTestWarningHook;
+use PHPUnit\Runner\BaseTestRunner;
+use PHPUnit\Runner\TestResultCache;
+
+final class ResultCacheExtension implements AfterSuccessfulTestHook, AfterSkippedTestHook, AfterRiskyTestHook, AfterIncompleteTestHook, AfterTestErrorHook, AfterTestWarningHook, AfterTestFailureHook, AfterLastTestHook
```

Comment:
The `Framework` directory is about "writing tests with PHPUnit". Classes related to "running tests with PHPUnit" belong in `Runner`.

Reply:
Just to confirm, you would like `ResultCacheExtension` to be moved to `Runner`, right? No problem, I will move it around!

Reply:
Yes, please. Thanks!

Reply:
Done!

Reply:
@epdenouden , `ResultCacheExtensionTest.php` shall also change a folder ;)